### PR TITLE
REF: use public indexers in groupby.ops

### PR DESCRIPTION
**Pull request by jbrockmendel**

### CODE REVIEW

File: pandas/core/groupby/ops.py [hunk 1]
```diff
@@ -780,7 +780,11 @@ def get_iterator(self, data: FrameOrSeries, axis: int = 0):
         Generator yielding sequence of (name, subsetted object)
         for each group
         """
-        slicer = lambda start, edge: data._slice(slice(start, edge), axis=axis)
+        if axis == 0:
```
Comment (jreback) [inline pandas/core/groupby/ops.py]: a more natural idiom (used in many places is)
```
indexer = [None] * 2
indexer[axis] = slice(start, edge)
slicer = data.iloc[indexer]
```

**Review by jreback — CHANGES_REQUESTED** [review]

**Review by WillAyd — COMMENTED** [review]
Lgtm outside of Jeff’s comment(s)

File: pandas/core/groupby/ops.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/groupby/ops.py]: i guess, but thats not really conducive to being a lambda

Comment (jreback) [reply pandas/core/groupby/ops.py]: why does this need to be a lamba?

Comment (jbrockmendel) [reply pandas/core/groupby/ops.py]: doesnt have to be, thats just the pattern we have now.  i find `iloc[:, start:end]` clearer than the 3-line version

Comment (jreback) [reply pandas/core/groupby/ops.py]: ok

**Review by jreback — APPROVED** [review]

**Pull request merged**